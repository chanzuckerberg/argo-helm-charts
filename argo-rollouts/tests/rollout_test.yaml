# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: test rollouts
templates:
  - rollout.yaml
tests:
  - it: should not create a rollout when deployment kind is not set to rollout
    set:
      global:
        deploymentKind: Deployment

    asserts:
      - hasDocuments:
          count: 0

  - it: should create a rollout when deployment kind is set to rollout
    set:
      global:
        deploymentKind: Rollout
      services:
        service1:
          shareProcessNamespace: false 
          podSecurityContext: {}
          securityContext: {}
          livenessProbe:
            failureThreshold: 3
            httpGet:
              path: / 
              port: http
              scheme: HTTP
            periodSeconds: 10 
            successThreshold: 1 
            timeoutSeconds: 1
            initialDelaySeconds: 30
          readinessProbe: 
            failureThreshold: 3
            httpGet: 
              path: /
              port: http 
              scheme: HTTP
            periodSeconds: 10
            successThreshold: 1 
            timeoutSeconds: 1 
            initialDelaySeconds: 30
          resources:
            limits:
              cpu: "1"
              memory: "1Gi"
            requests:
              cpu: "100m" 
              memory: "128Mi"
          image:
            repository: nginx 
            pullPolicy: IfNotPresent
            tag: "latest"
          service:
            type: ClusterIP
            port: 80
          startupProbe:
            enabled: true
          persistence:
            enabled: false
            pvc:
              accessModes:
                - ReadWriteOnce
              resources:
                requests:
                  storage: 8Gi
              storageClassName: "default"
              dataSource:
                name: ""
                kind: ""
                apiGroup: "snapshot.storage.k8s.io/v1"
            existingClaim: ""
            mountPath: ""
          appContext:
            envContextConfigMapName: ""
            stackContextConfigMapName: ""
          appSecrets:
            envSecret:
              secretName: ""
              secretKey: ""
            stackSecret:
              secretName: ""
              secretKey: ""
            clusterSecret:
              secretName: ""
              secretKey: ""
          args: ["arg1", "arg2"]
          command: ["command1", "command2"]
          serviceAccount:
            create: true
          autoscaling:
            enabled: true
            minReplicas: 2
            maxReplicas: 10
            maxUnavailable: 1
          replicaCount: 2
          sidecars:
            - name: sidecar1
              image: "sidecar1:latest"
            - name: sidecar2
              image: "sidecar2:latest"
          initContainers:
            - name: initContainer1
              image: "alpine:latest"
              command: ["echo", "Hello World"]
      rolloutAnalysis:
        analysisRunMetadata:
          annotations:
            test: "value"
        strategy:
          bluegreen:
            autoPromotionEnabled: true
            prePromotionAnalysis:
              templates:
                - templateName: "template1"
                - templateName: "template2"
              args:
                - name: "arg1"  
                  value: "value1"
              analysisRunMetadata:
                annotations:
                  test: "value"

    asserts:
      # --- manifest should match snapshot ---
      - matchSnapshot: {}
      # --- API and Metadata Checks ---
      - hasDocuments:
          count: 1
      - isKind:
          of: Rollout
      - equal:
          path: metadata.name
          value: release-name-argo-rollouts-service1
      - equal:
          path: metadata.annotations["argocd.argoproj.io/sync-options"]
          value: ServerSideApply=false
      - equal:
          path: metadata.annotations["test"]
          value: value
      
      # --- Basic Rollout Spec Checks ---
      - equal:
          path: spec.replicas
          value: 2
      - equal:
          path: spec.revisionHistoryLimit
          value: 3
      - equal:
          path: spec.selector.matchLabels["app.kubernetes.io/service"]
          value: release-name-argo-rollouts-service1