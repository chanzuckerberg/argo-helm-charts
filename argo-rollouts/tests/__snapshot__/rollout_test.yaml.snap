should create a rollout when deployment kind is set to rollout:
  1: |
    apiVersion: argoproj.io/v1alpha1
    kind: Rollout
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: ServerSideApply=false
        test: value
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: argo-rollouts
        app.kubernetes.io/service: release-name-argo-rollouts-service1
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: argo-rollouts-0.1.0
      name: release-name-argo-rollouts-service1
    spec:
      replicas: 2
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/service: release-name-argo-rollouts-service1
      strategy:
        bluegreen:
          activeService: release-name-argo-rollouts-service1
          autoPromotionEnabled: true
          prePromotionAnalysis:
            analysisRunMetadata:
              annotations:
                test: value
            args:
              - name: arg1
                value: value1
              - name: target-service-name
                value: release-name-argo-rollouts-service1
              - name: target-service-namespace
                value: NAMESPACE
              - name: preview-service-name
                value: release-name-argo-rollouts-service1-preview
              - name: preview-rs-hash
                valueFrom:
                  podTemplateHashValue: Latest
            templates:
              - templateName: template1
              - templateName: template2
          previewService: release-name-argo-rollouts-service1-preview
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: argo-rollouts
            app.kubernetes.io/service: release-name-argo-rollouts-service1
            app.kubernetes.io/version: 1.16.0
            helm.sh/chart: argo-rollouts-0.1.0
        spec:
          containers:
            - args:
                - arg1
                - arg2
              command:
                - command1
                - command2
              env: []
              image: nginx:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 30
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              name: argo-rollouts
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 30
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: "1"
                  memory: 1Gi
                requests:
                  cpu: 100m
                  memory: 128Mi
              securityContext: {}
              startupProbe: {}
            - env: []
              image: sidecar1:latest
              name: sidecar1
            - env: []
              image: sidecar2:latest
              name: sidecar2
          dnsPolicy: ClusterFirst
          initContainers:
            - command:
                - echo
                - Hello World
              env: []
              image: alpine:latest
              name: initContainer1
          restartPolicy: Always
          securityContext: {}
          serviceAccountName: release-name-argo-rollouts-service1
          shareProcessNamespace: false
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/service: release-name-argo-rollouts-service1
              maxSkew: 2
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: ScheduleAnyway
